### DISCUSSION TITLE
Trouble installing Gemini IRAF Data Reduction Software (pyraf)

### DISCUSSION THREAD

**Discussion by hatpiano** [iraf-community/iraf#404] [Q&A] [answered]
Hello everyone, I am having a hard time trying to get the Gemini IRAF Date Reduction Software installed properly. I'm using the WSL and Ubuntu 22.04. The website I used to download IRAF was [this one](https://iraf-community.github.io/install.html). I was able to get this working just fine. The problem was when I tried to follow the instruction on [this website](https://www.gemini.edu/observing/phase-iii/reducing-data/gemini-iraf-data-reduction-software).

I managed to make it to the step where I have to input "pyraf" with the geminiconda environment active. No matter what I seem to do, anytime I try and run the command "pyraf" with the geminiconda environment active, I get a broken pipe error. I'm not sure what the error means or how to fix it, or even what I'm doing wrong. I am following the steps perfectly I think. I even uninstalled my WSL and reinstalled it but it still didn't work. 

If anyone has any advice, I would appreciate it greatly. Attached is a photo of the error message (the error message is large so I had to use 2 screenshots).

![error screenshot 1](https://github.com/user-attachments/assets/2e01847e-341c-48d4-b748-59df21d59aa9)
![error screenshot 2](https://github.com/user-attachments/assets/78c69c95-ff0b-4d90-bc54-494bcdd7fa79)

**Comment by olebole** [comment, marked as answer]
I think the best way to handle problems with Gemini is to ask the Gemini people, as they have the best experience with the installation of their software. Specifically, as written on the page you linked to it requires the 32-bit IRAF version from Miniforge or Anaconda, not the 64-bit version from iraf-community.github.io. Keep in mind that WSL doesn't allow to run 32-bit binaries, so you would probably also not have success by following their instructions. You need a 64-bit version of Gemini for WSL.

Generally, your error one sees in your screenshot suggests that it tries to run `x_system.e` from geminiconda. `x_system.e` is however an executable of the IRAF core, and is (if you followed the instructions on iraf-community) under `/usr/local/lib/iraf/bin/`. So, if you really want to use the self-compiled comunity IRAF, you should make sure that the `iraf` environment variable points to `/usr/local/lib/iraf/`. However, as I said you still need a 64-bit  version of Gemini, and of the other dependencies (mainly `stsdas`, or its replacement `st4gem`). Unless they are available, you will probably out of luck here, and this all requires a lot of manual tweaking.

NOIRLAB made a few months ago a port of Gemini to 64 bit, including the required `stsdas` (as `st4gem`) package and the required changes in the IRAF core: https://iraf.noirlab.edu. You may try to download this and see if this works in WSL.

**Reply by hatpiano** [reply]
Thank you for the reply! I managed to get everything working after creating a VM. Thanks for pointing out what was going wrong!
